### PR TITLE
Support for aiohttp >= 2.x

### PR DESCRIPTION
**Pull request by evertlammerts**
This hopefully solves #15. It finishes what @thecodingrobot started in #19. Tests are passing.

### CODE REVIEW

**Comment by evertlammerts** [comment]
Ping @HonzaKral @karmi Any feedback? Can we get this in?

On a related note: I'm shipping a patched version with my own app now, but I'm having doubts about the status of the library. Is Elastic going to maintain this or should I just use HTTP / the aiohttp client for async elasticsearch?

**Comment by bamthomas** [comment]
I'm also interested by this PR. When will it be merged ?

File: elasticsearch_async/connection.py [hunk 1]
```diff
@@ -29,7 +29,7 @@ def __init__(self, host='localhost', port=9200, http_auth=None,
             connector=aiohttp.TCPConnector(
                 loop=self.loop,
                 verify_ssl=verify_certs,
-                conn_timeout=self.timeout,
+                keepalive_timeout=self.timeout,
```
Comment (HonzaKral) [inline elasticsearch_async/connection.py]: I don't think we want to change the `keepalive_timeout` - we should leave the `conn_timeout`

File: setup.py [hunk 2]
```diff
@@ -16,10 +16,10 @@
...
 
 tests_require = [
-    'pytest',
-    'pytest-asyncio',
-    'pytest-cov',
-    'pytest-catchlog',
+    'pytest==3.1.2',
+    'pytest-asyncio==0.6.0',
+    'pytest-cov==2.5.1',
+    'pytest-catchlog==1.2.2',
```
Comment (HonzaKral) [inline setup.py]: please do not lock to the build version - those are typically bug fixes etc which we should follow.

File: test_elasticsearch_async/test_connection.py [hunk 3]
```diff
@@ -58,7 +58,7 @@ def test_error_is_properly_logged(connection, caplog, port, server):
 def test_timeout_is_properly_raised(connection, server):
     @asyncio.coroutine
     def slow_request():
-        yield from asyncio.sleep(0.01)
+        yield from asyncio.sleep(1)
```
Comment (HonzaKral) [inline test_elasticsearch_async/test_connection.py]: that this change is needed is very suspicious, the sleep is 2 orders of magnitude larger than the timeout and it still doesn't work - something is wrong here - maybe the timeout handling in connection that this is testing.

Comment (bamthomas) [inline test_elasticsearch_async/test_connection.py]: it is due to the `TimeoutHandle` in aiohttp that uses ceil : 

    def start(self):
        if self._timeout is not None and self._timeout > 0:
            at = ceil(self._loop.time() + self._timeout)
            return self._loop.call_at(at, self.__call__)

The `ClientSession` requests timeout is called `read_timeout`.

Do you want another PR ?

Comment (evertlammerts) [inline test_elasticsearch_async/test_connection.py]: `read_timeout` is also a `TimeoutHandle`. From [client.py](https://github.com/aio-libs/aiohttp/blob/master/aiohttp/client.py#L206):

```
tm = TimeoutHandle(
    self._loop,
    timeout if timeout is not sentinel else self._read_timeout)
```

I'll look into this asap.

File: elasticsearch_async/connection.py (hunk 1 above)
Comment (evertlammerts) [inline elasticsearch_async/connection.py]: `conn_timeout` is no longer there, the docs on `TCPConnector` are a out of date. But you're right we shouldn't use `keepalive_timeout` here, that makes no sense as a replacement. The only other timeout we can use is `ClientSession.read_timeout`, which is cumulative. But that seems to be rounded off to seconds. I'll dig a little deeper soon.

Comment (HonzaKral) [inline elasticsearch_async/connection.py]: It has moved to `ClientSession` - http://aiohttp.readthedocs.io/en/stable/client_reference.html#aiohttp.ClientSession

`keepalive_timeout` is entirely different functionality and cannot be used for this use case.

File: test_elasticsearch_async/test_connection.py (hunk 3 above)
Comment (HonzaKral) [inline test_elasticsearch_async/test_connection.py]: It would be ideal to only have this PR touch the parts needed to upgrade to aiohttp >= 2.x and not touch other parts of the code or constructs.

We can then have separate PRs for those if those are not ideal, but one step at a time please. This is changing too many things at once.

**Comment by HonzaKral** [comment]
this has been fixed in 4a0f82129bdd8521ea4449b3f73fdba8adcc5a1d
